### DISCUSSION TITLE
v1.0.1

### DISCUSSION THREAD

**Discussion by SHAYKHUL** [SHAYKHUL/Pysurf#2] [Ideas]
# Browser Application - Version 1.0.1 (v1) Updates

### 1. **Tabbed Browsing**
   - **Multiple tabs** can now be opened, each with its own webpage.
   - Tabs can be **closed** and **dynamically added**.
   - Provides an overall improved multitasking experience.

### 2. **Improved User Interface (UI)**
   - Expanded **window size** for better usability.
   - A more **user-friendly layout** was introduced.
   - The **navigation bar** now includes more intuitive controls.

### 3. **Integrated Navigation Bar**
   - Combined **address bar** and navigation buttons (Back, Forward, Reload, Home) into a streamlined interface.
   - Users can **navigate** with a more intuitive and efficient design.

### 4. **Dark Mode Support**
   - **Toggle dark mode** for a visually enhanced browsing experience, ideal for low-light environments.

### 5. **Customizable Search Engine**
   - Users can now choose between **Google**, **Bing**, or **DuckDuckGo** directly from the navigation bar for a more personalized search experience.

### 6. **Bookmarks Feature**
   - Added **bookmarking** functionality, allowing users to **save** and **access** their favorite websites easily.

### 7. **Browsing History**
   - A **history feature** has been implemented, enabling users to quickly revisit previously visited websites.

### 8. **Session Management**
   - **Session state** is automatically saved and restored, keeping track of previously opened tabs for a seamless browsing experience on startup.

**Full Changelog**: https://github.com/SHAYKHUL/Pysurf/commits/v1.0.1

<hr /><em>This discussion was created from the release <a href='https://github.com/SHAYKHUL/Pysurf/releases/tag/'>v1.0.1</a>.</em>

**Comment by sesawan** [comment]
This app is too buggy and something not works well
